src/snowflake/snowpark/modin/pandas/index.py

```diff
@@ -0,0 +1,153 @@
+    Index([1, 2, 3], dtype='uint8')
+    """
+
+    # same fields as native pandas index constructor
+    def __init__(
+        self,
+        data=None,
+        dtype=None,
+        copy=False,
+        name=None,
+        tupleize_cols=True,
+        query_compiler=None,
+    ) -> None:
+        self._index = None
+
+        # TODO: SNOW-1359041: Switch to lazy index implementation
+        if isinstance(data, (pd.DataFrame, pd.Series)):
+            qc = data._query_compiler
+        elif isinstance(data, pd.Index):
+            self = data
+            return
+        elif query_compiler is not None:
+            qc = query_compiler
+        else:
+            index_as_native_df = native_pd.Index(
+                data=data,
+                dtype=dtype,
+                copy=copy,
+                name=name,
+                tupleize_cols=tupleize_cols,
+            ).to_frame()
+            from snowflake.snowpark.modin.plugin.compiler.snowflake_query_compiler import (
+                SnowflakeQueryCompiler,
+            )
+
+            qc = SnowflakeQueryCompiler.from_pandas(index_as_native_df)
+
+        self._query_compiler = qc if qc else None
+
+    @property
+    def _column_names(self):
+        return self._query_compiler.get_index_names()
+
+    @property
+    def _ordered_dataframe(self):
+        return self._query_compiler._modin_frame.ordered_dataframe
+
+    @property
+    def _snowflake_quoted_identifiers(self):
+        return (
+            self._query_compiler._modin_frame.index_column_snowflake_quoted_identifiers
+        )
+
+    @property
+    def is_multi_index(self, axis=0):
+        return self._query_compiler.is_multiindex(axis=axis)
+
+    @property
+    def values(self):
+        from snowflake.snowpark.modin.plugin._internal.utils import (
+            snowpark_to_pandas_helper,
+        )
+
+        values = snowpark_to_pandas_helper(
+            self._ordered_dataframe.select(self._snowflake_quoted_identifiers)
+        ).values
+
+        if self.is_multi_index:
+            tuples = [tuple(v) for v in values]
+            values = np.empty(len(tuples), dtype=object)
+            values[:] = tuples
+        else:
+            values = values.flatten()
+        return values
+
+        return snowpark_to_pandas_helper(
+            self._ordered_dataframe.select(self._snowflake_quoted_identifiers)
+        ).values.flatten()
+
+    @property
+    def names(self):
+        return self._column_names
+
+    @property
+    def name(self):
+        return self.names[0]
+
+    def to_pandas(self):
+        return self._query_compiler.index
+
+    def __array__(self, dtype=None) -> np.ndarray:
+        """
+        The array interface, return my values.
+        """
+        return self.to_pandas().__array__(dtype=dtype)
+
+    def __repr__(self):
+        return self.to_pandas().__repr__()
+
+    def __iter__(self):
+        # TODO: SNOW-1359041: Do we need to update this in the next PR to not call to_pandas()
```

Comment:
nit: If you're leaving a TODO for the future (you will address it in a different PR) add more context to the task. Does not have to be long as long as the ticket has details.